cmake/Helpers.cmake

```diff
@@ -593,86 +492,73 @@ endif()
-        INCLUDE(platforms/vita/vita.cmake)
-    ENDIF(VITA)
+    INCLUDE(platforms/vita/vita.cmake)
 ENDFUNCTION()
 
 FUNCTION(CONFIGURE_UNIX_SPECIFICS)
-    IF(UNIX)
-        SET(CMAKE_THREAD_PREFER_PTHREAD true)
-        SET(CMAKE_THREAD_PREFER_PTHREAD ${CMAKE_THREAD_PREFER_PTHREAD} CACHE INTERNAL "")
-        FIND_PACKAGE(Threads REQUIRED)
-
-        # TODO ONLY FOR UBUNTU
-        # also put the chosen paths in the man page (Ubuntu)
-        CONFIGURE_FILE(
-            "${CMAKE_CURRENT_SOURCE_DIR}/gemrb.6.in"
-            "${CMAKE_CURRENT_BINARY_DIR}/gemrb.6"
-            IMMEDIATE @ONLY
-        )
-    ENDIF(UNIX)
+    SET_INTERNAL(CMAKE_THREAD_PREFER_PTHREAD true)
+    FIND_PACKAGE(Threads REQUIRED)
+
+    CONFIGURE_FILE(
+        "${CMAKE_CURRENT_SOURCE_DIR}/gemrb.6.in"
+        "${CMAKE_CURRENT_BINARY_DIR}/gemrb.6"
+        IMMEDIATE @ONLY
+    )
 ENDFUNCTION()
 
 FUNCTION(CONFIGURE_RPI_SPECIFICS)
     # check for RaspberryPi
-    IF(UNIX)
-        FIND_FILE(RPI NAMES bcm_host.h PATHS "/opt/vc/include")
-        # By default, Pi0 to Pi3 models use the legacy (Broadcom) GLESv2 drivers, from /opt/vc.
-        # Newer models (Pi4) don't support it, using the open source MESA drivers.
-        # NOTE: the Pi3B(+) models can also run with open source MESA drivers, but support for it must be explicitely enabled
-        IF(RPI AND NOT DISABLE_VIDEOCORE)
-            SET(HINT_GLES_LIBNAME brcmGLESv2)
-            SET(HINT_GLES_LIBNAME ${HINT_GLES_LIBNAME} CACHE INTERNAL "")
-
-            SET(HINT_GLES_INCDIR "/opt/vc/include")
-            SET(HINT_GLES_INCDIR ${HINT_GLES_INCDIR} CACHE INTERNAL "")
-
-            SET(HINT_GLES_LIBDIR "/opt/vc/lib")
-            SET(HINT_GLES_LIBDIR ${HINT_GLES_LIBDIR} CACHE INTERNAL "")
-
-            SET(OPENGL_BACKEND GLES)
-            SET(OPENGL_BACKEND ${OPENGL_BACKEND} CACHE INTERNAL "")
-
-            SET(SDL_BACKEND SDL2)
-            SET(SDL_BACKEND ${SDL_BACKEND} CACHE INTERNAL "")
-        ENDIF()
+    FIND_FILE(RPI NAMES bcm_host.h PATHS "/opt/vc/include")
+    # By default, Pi0 to Pi3 models use the legacy (Broadcom) GLESv2 drivers, from /opt/vc.
+    # Newer models (Pi4) don't support it, using the open source MESA drivers.
+    # NOTE: the Pi3B(+) models can also run with open source MESA drivers, but support for it must be explicitely enabled
+    IF(RPI AND NOT DISABLE_VIDEOCORE)
+        SET_INTERNAL(HINT_GLES_LIBNAME brcmGLESv2)
+        SET_INTERNAL(HINT_GLES_INCDIR "/opt/vc/include")
+        SET_INTERNAL(HINT_GLES_LIBDIR "/opt/vc/lib")
+        SET_INTERNAL(OPENGL_BACKEND GLES)
+        SET_INTERNAL(SDL_BACKEND SDL2)
     ENDIF()
 ENDFUNCTION()
 
 FUNCTION(CONFIGURE_TARGET_PLATFORM_SPECIFICS)
-    #EACH OF THEM DOES NOTHING IF TARGET PLATFORM IS NOT MATCHING
-    CONFIGURE_APPLE_SPECIFICS()
-    CONFIGURE_VITA_SPECIFICS()
-    CONFIGURE_UNIX_SPECIFICS()
-    CONFIGURE_RPI_SPECIFICS()
+    SET_INTERNAL(BAKE_ICON 0)
+    IF(APPLE)
+        CONFIGURE_APPLE_SPECIFICS()
+        RETURN()
```

Comment:
These returns are bad. At a glance they're only redundant, saving a few checks, but actually the blocks are not independant — UNIX also matches on APPLE.

Reply:
Ok so what did you mean with returns previously?
I rarely use returns in CMake so I don't see profits of them at all.

Reply:
They would make sense if they were in the subfunctions themselves, not here. This way is cleaner though, so I suggest you just remove them.

Reply:
Now I get it. I might have done it as an early exits... <facepalm>
But yeah, I will remove them.